packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -187,14 +191,40 @@ const App = (props) => {
                 body: {}
             })
         }
+    }, [baskets])
+
+    useEffect(() => {
         // update the basket currency if it doesn't match the current locale currency
-        if (baskets?.baskets?.[0]?.currency && baskets.baskets[0].currency !== currency) {
+        if (basket?.currency && basket?.currency !== currency) {
             updateBasket.mutate({
-                parameters: {basketId: baskets.baskets[0].basketId},
+                parameters: {basketId: basket.basketId},
                 body: {currency}
             })
         }
-    }, [baskets])
+
+        // update the basket customer email
+        if (
+            basket &&
+            customer?.isRegistered &&
+            customer?.customerId === basket?.customerInfo?.customerId &&
+            customer?.email &&
+            customer?.email !== basket?.customerInfo?.email
+        ) {
+            updateCustomerForBasket.mutate({
+                parameters: {basketId: basket.basketId},
+                body: {
+                    email: customer?.email
+                }
+            })
+        }
+    }, [
+        basket?.currency,
+        basket?.customerInfo?.customerId,
+        customer?.isRegistered,
+        customer?.customerId,
+        customer?.email,
+        basket?.customerInfo?.email
```

Comment:
We are tracking a lot of dependencies here. Why do we need to track so many values here? 🤔 
The currency logic does not seem to be related with the rest. Can we separate it into two useEffect 
```
useEffect(() => {}, [basket?.currency])
useEffect(() => {}, [(the rest of dependencies)]
```

Reply:
Is there a case where the customer.customerId and basket.customerInfo.customerId is different? I assume the backend should have those values in sync whenever you create a basket with a customer Id

Reply:
I had taken the `customer.customerId === basket.customerInfo.customerId` condition from the v2 version of this code. But you're correct that I don't think there's a case where the two would be different since useCurrentBasket uses customerId to lookup the basket. I'll remove that condition since that lets us remove 2 dependencies.

Reply:
Update: currency and email updates have been separated into 2 separate useEffects